GiEnJul/Controllers/MunicipalityController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using AutoMapper;
+using GiEnJul.Auth;
+using GiEnJul.Dtos;
+using GiEnJul.Repositories;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace GiEnJul.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class MunicipalityController: ControllerBase
+    {
+        private readonly IMunicipalityRepository _municipalityRepository;
+        private readonly IMapper _mapper;
+
+        public MunicipalityController(IMunicipalityRepository municipalityRepository, IMapper mapper)
+        {
+            _municipalityRepository = municipalityRepository;
+            _mapper = mapper;
+        }
+
+        [HttpDelete]
+        [Authorize(Policy = Policy.SuperAdmin)]
+        public async Task<ActionResult> DeleteSingleContent([FromBody] DeleteMunicipalityDto content)
+        {
+            try
+            {
+                await _municipalityRepository.DeleteEntry(_mapper.Map<Models.Municipality>(content));
+                return Ok();
+            }
+            catch
+            {
+                return BadRequest("Could not delete entry");
+            }
+        }
+
+        [HttpPost]
+        [Authorize(Policy = Policy.SuperAdmin)]
+        public async Task<ActionResult> PostContent([FromBody] PostMunicipalityDto content)
+        {
+            var names = await _municipalityRepository.GetAll();
+            if (names.Where(x => x.RowKey == content.Name).Any())
+                return BadRequest("RowKey already exists, use put request to edit");            
+            
+            await _municipalityRepository.InsertOrReplaceAsync(_mapper.Map<Models.Municipality>(content));  
+            return Ok();
+        }
+
+        [HttpPut]
+        [Authorize(Policy = Policy.UpdateMunicipality)]
+        public async Task<ActionResult> PutContent([FromBody] PostMunicipalityDto content)
+        {
+            var names = await _municipalityRepository.GetAll();
+            if (!names.Where(x => x.RowKey == content.Name).Any())
```

Comment:
```suggestion
            var entities = await _municipalityRepository.GetAll();
            var exsistingMunicipalities = _mapper.Map<List<Models.Municipality>>(entities);
            if (!exsistingMunicipalities.Any(x => x.Name == content.Name))
```
.Any() can actually have a filter so we don't need to use Where() here 👍